### DISCUSSION TITLE
[Feature Request] Vertical table or responsive table design for small screen devices

### DISCUSSION THREAD

**Discussion by ChaserZ98** [heroui-inc/heroui#4731] [Feature requests]
### Is your feature request related to a problem? Please describe.

The current table component does not adapt well on the situation where you have plenty columns of data in a small screen. It is difficult for users to have a full view of one single row of data in a device with small screen. Such bad experience also happens when a column requires much more height than other columns to display. For example, an image column with other short text columns. 

### Describe the solution you'd like

Add an `isVertical` attribute to Table props to change table from displaying rows of data to columns of data. And if `isHeaderSticky` is passed, the header should be sticky to the horizontal scroll instead of vertical scroll.
```jsx
<Table aria-label="Example empty table" isVertical>
  <TableHeader>
    <TableColumn>Column 1</TableColumn>
    <TableColumn>Column 2</TableColumn>
    <TableColumn>Column 3</TableColumn>
  </TableHeader>
</Table>
```
An expected behavior would be as the following image shows.
![68747470733a2f2f7777772e6e6963686570757273756974732e636f6d2f77702d636f6e74656e742f75706c6f6164732f323031392f30332f636f6d70617269736f6e7461626c652e706e67](https://github.com/user-attachments/assets/759f69be-576b-4860-869a-e4e930d41640)

### Describe alternatives you've considered

Another alternative would be something like [table-with-header-cells-in-the-first-column-only](https://www.w3.org/WAI/tutorials/tables/one-header/#table-with-header-cells-in-the-first-column-only). But I doubt if it fits the current design logic of Table component in NextUI.

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1405/[feature-request]-vertical-table-or-responsive-table-design-for-small">ENG-1405 [Feature Request] Vertical table or responsive table design for small screen devices</a></p>

**Comment by qwertyroop** [comment]
I will be working on this kindly assign me this.

**Comment by wingkwong** [comment]
@qwertyroop please discuss your plan with the team in discord first. You may ping me at discord (same id).

**Comment by ChaserZ98** [comment]
@qwertyroop @wingkwong Is it in progress or no one is currently working on it?
